### ISSUE TITLE
hivemind.dht nice-to-haves

### ISSUE THREAD

**Issue by justheuristic** [learning-at-home/hivemind#56]
* [x] bulk traverse_dht - implement a search algorithm that looks for several queries in parallel, reusing search metadata ( #53 )
* [ ] channel reuse - DHTProtocol creates a new channel on every run. This is inefficient for consecutive requests (e.g. find -> store)
* [x] congestion - #51 broke max_concurrent rpc limit, @justheuristic has sworn to fix that ( #55 )
* [ ] proactive caching - add an option for DHTNode to automatically re-get k most recently used keys that approach expiration
* [x] use uvloop ( #55 )
* [x] efficient routing table strucure for get_nearest_neighbors ( #60 )
* [x] update benchmark_dht
   * [ ] tune channel params on the benchmark
   * [ ] tune bucket size and caching

Extras:
* add fault tolerance test
* notify peers on shutdown
